### PR TITLE
Frenchincome

### LINKED ISSUE #276: FR_income task missing
Not to be found in the repo, we have to find the code.

### PR DESCRIPTION
**Pull request by antoniocarlon**
Fixes #276 

- Added makefile target
- Found new data source for FR income
- Solved importing issues
- Updated tags

### CODE REVIEW

**Comment by ethervoid** [comment]
Before making the CR please solve the conflicts and the tests (merging/rebasing master should do the trick)

File: tasks/fr/fr_income.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            'sourcetag': SourceTags()
+        }
+
+
+    def version(self):
+        return 1
+
+    def columns(self):
+        cols = OrderedDict()
+        input_ = self.input()
```
Comment (ethervoid) [inline tasks/fr/fr_income.py]: This is related to an unmerged change PR #150 , if we're going to do it this way we should merge that PR

File: tasks/fr/fr_income.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                denominators = denominators.split(',')
+                targets_dict = {}
+                for x in denominators:
+                    x = x.strip()
+                    targets_dict[cols.get(x)] = 'denominator'
+                targets_dict.pop(None, None)
+                universes = universe.split(',')
+                for x in universes:
+                    x = x.strip()
+                    targets_dict[cols.get(x)] = 'universe'
```
Comment (ethervoid) [inline tasks/fr/fr_income.py]: Becuase `universe` and `denominator` are mutually exclusive we should improve this part and make it more readable for the future

File: tasks/fr/fr_income.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                targets_dict.pop(None, None)
+                cols[var_code] = OBSColumn(
+                    id=var_code,
+                    type='Numeric',
+                    name=long_name,
+                    description =long_name,
+                    # Ranking of importance, sometimes used to favor certain measures in auto-selection
+                    # Weight of 0 will hide this column from the user.  We generally use between 0 and 10
+                    weight=5,
+                    aggregate='',
```
Comment (ethervoid) [inline tasks/fr/fr_income.py]: weird we don't have aggregate :\

File: tasks/fr/fr_income.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+                    # it's relevant to, and which section(s) of the catalog it should appear in
+                    tags=[france, unittags[var_unit]],
+                    targets= targets_dict
+                )
+                subsections = subsections.split(',')
+                for s in subsections:
+                    s = s.strip()
+                    subsection_tag = subsectiontags[s]
+                    cols[var_code].tags.append(subsection_tag)
+                if var_unit in ('households', 'people', 'tax_consumption_units'):
```
Comment (ethervoid) [inline tasks/fr/fr_income.py]: Move this to a constant array

File: tasks/fr/fr_income.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+                    cols[var_code].aggregate = 'sum'
```
Comment (ethervoid) [inline tasks/fr/fr_income.py]: Oh here is the aggregate xD

File: tasks/fr/fr_income.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+                        'SELECT {ids_typed} '
+                        'FROM {input} '.format(
+                            ids=colnames,
+                            ids_typed=colnames_typed,
+                            output=self.output().table,
+                            input=self.input()['data'].table
+                        ))
+
+
+class IRISIncomeTables(WrapperTask):
```
Comment (ethervoid) [inline tasks/fr/fr_income.py]: Do we need a meta wrapper? I think at least to test this part automatically is needed

File: tasks/fr/fr_income.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        requirements = {
+            'data': RawIncomeIrisData(table_theme=self.table_theme),
+            'meta': IrisIncomeColumns(table_theme=self.table_theme),
+            'geometa': OutputAreaColumns(),
+        }
+        return requirements
+
+    def columns(self):
+        cols = OrderedDict()
+        cols['IRIS'] = self.input()['geometa']['dcomiris']
```
Comment (ethervoid) [inline tasks/fr/fr_income.py]: this shouldn't be named `dcomiris` I see in other columns task they keep the name

**Comment by antoniocarlon** [comment]
Extracted `fr-geo` as a new target.
Note that Acceptance should test `fr-income`, `fr-insee` and `fr-geo`.

**Pull request merged**